### PR TITLE
Auto Detect Monitor Refresh Rate

### PR DESCRIPTION
**Pull request by orcephrye**
Added changes to config.c and main.c to support the ability auto-detect the monitor refresh rate which is then doubled for the purpose of the FPS Limit.

### CODE REVIEW

**Comment by gnif** [comment]
Great work, please though squash the 2nd commit.

File: client/src/main.c [hunk 1]
```diff
@@ -1120,6 +1120,22 @@ int run()
...
   updatePositionInfo();
 
+  //Auto detect active monitor refresh rate for FPS Limit if no FPS Limit was passed.
+  if (params.fpsLimit == -1)
+  {
+      SDL_DisplayMode current;
+      if (SDL_GetCurrentDisplayMode(SDL_GetWindowDisplayIndex(state.window), &current) == 0)
+      {
+          state.frameTime = 1e9 / ((int) current.refresh_rate * 2);
+      } else {
```
Comment (gnif) [inline client/src/main.c]: braces on separate lines please

File: client/src/main.c [hunk 2]
```diff
@@ -1120,6 +1120,22 @@ int run()
...
+          DEBUG_WARN("Unable to capture monitor refresh rate using the default FPS Limit: 200");
+          state.frameTime = 1e9 / 200;
+      }
+  } else {
```
Comment (gnif) [inline client/src/main.c]: braces on separate lines please

File: client/src/main.c [hunk 3]
```diff
@@ -1000,7 +1000,7 @@ int run()
   state.running   = true;
   state.scaleX    = 1.0f;
   state.scaleY    = 1.0f;
-  state.frameTime = 1e9 / params.fpsLimit;
+  //state.frameTime = 1e9 / params.fpsLimit;
```
Comment (gnif) [inline client/src/main.c]: don't comment, please remove

**Comment by orcephrye** [comment]
I have done as requested.

**Comment by gnif** [comment]
Sorry, one final change I missed, please re-title your commit message:

    [c-host] autodetect monitor refresh rate for fps limit

For future reference for this and most projects refrain from a commit message that exceeds 80 characters, if you need more room to expand on what the commit does place it after the first line wrapping at 80 characters wide.

**Comment by orcephrye** [comment]
Ok force pushed an amend to the commit.

**Pull request merged**